File: frontend/src/types/email.ts
export type EmailCategory =
  | 'important' | 'normal' | 'promotion' | 'bill' | 'school_work' | 'needs_reply' | 'spam';

export interface EmailResponse {
  id: number;
  message_id: string;
  sender: string;
  recipients: string;
  subject: string;
  body: string;
  received_at: string;
  is_read: boolean;
  category: string;
  importance_score: number;
  summary: string | null;
  ai_processed: boolean;
  spam_confidence: number | null;
  spam_signals: string | null;
  imported_source: string;
  created_at: string;
  updated_at: string;
}

export interface EmailListResponse {
  items: EmailResponse[];
  total: number;
  page: number;
  page_size: number;
}

export interface AIError {
  message: string;
  type: 'auth_error' | 'rate_limit' | 'server_error' | 'timeout' | 'provider_error' | string;
  retryable: boolean;
}

export interface EmailDetailResponse extends EmailResponse {
  drafts: DraftResponse[];
  reminders: ReminderResponse[];
}

export interface ClassifyResponse {
  category: string;
  importance_score: number;
  error: AIError | null;
}

export interface SummarizeResponse {
  summary: string;
  error: AIError | null;
}

export interface GenerateDraftResponse {
  id: number;
  tone: string;
  content: string;
  error: AIError | null;
}

export interface ExtractRemindersResponse {
  reminders: ReminderItem[];
  error: AIError | null;
}

export interface ReminderItem {
  title: string;
  description: string | null;
  reminder_type: string;
  due_at: string | null;
}

// re-export from other type files
import { DraftResponse } from './draft';
import { ReminderResponse } from './reminder';

File: backend/app/schemas/email.py
from __future__ import annotations
import datetime
from typing import Optional
from pydantic import BaseModel, Field

from app.schemas import EmailCategory


class ImportResponse(BaseModel):
    imported: int
    skipped: int = 0
    errors: list[str] = []


class EmailBase(BaseModel):
    sender: str
    recipients: str
    subject: str
    body: str
    received_at: datetime.datetime


class EmailResponse(BaseModel):
    id: int
    message_id: str
    sender: str
    recipients: str
    subject: str
    body: str
    received_at: datetime.datetime
    is_read: bool
    category: str
    importance_score: int
    summary: Optional[str] = None
    spam_confidence: Optional[float] = None
    spam_signals: Optional[str] = None
    imported_source: str
    created_at: datetime.datetime
    updated_at: datetime.datetime

    model_config = {"from_attributes": True}


class EmailListResponse(BaseModel):
    items: list[EmailResponse]
    total: int
    page: int
    page_size: int


class EmailPatchRequest(BaseModel):
    is_read: Optional[bool] = None
    category: Optional[EmailCategory] = None
    importance_score: Optional[int] = Field(None, ge=1, le=5)


class EmailDetailResponse(EmailResponse):
    drafts: list[DraftResponse] = []
    reminders: list[ReminderResponse] = []


from app.schemas.draft import DraftResponse  # noqa: E402
from app.schemas.reminder import ReminderResponse  # noqa: E402
EmailDetailResponse.model_rebuild()
